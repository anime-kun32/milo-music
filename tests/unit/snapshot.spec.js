import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import SongItem from '@/components/SongItem.vue';

describe('Snpashots SongItem.vue', () => {
  it('renders correctly', () => {
    const song = {
      docID: 'abc',
      modified_name: 'test',
      display_name: '',
      comment_count: 5,
    };

    const wrapper = shallowMount(SongItem, {
      propsData: { song },
      global: {
        components: {
          'router-link': RouterLinkStub,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
