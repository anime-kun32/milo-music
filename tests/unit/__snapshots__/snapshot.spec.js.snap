// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snpashots SongItem.vue renders correctly 1`] = `
<li
  class="example flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-100"
  id="song-id-abc"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600 composition-item"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    >
      Artist: 
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      custom=""
      to="[object Object]"
    />
  </div>
</li>
`;
